unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/QuestsPanelHUD/QuestsPanelEntry.cs

```diff
@@ -0,0 +1,133 @@
+            pinnedQuests.OnAdded += OnPinnedQuests;
+            pinnedQuests.OnRemoved += OnUnpinnedQuest;
+        }
+
+        public void Populate(QuestModel newQuest)
+        {
+            quest = newQuest;
+
+            QuestTask incompletedTask = quest.sections.FirstOrDefault(x => x.progress < 1)?.tasks.FirstOrDefault(x => x.progress < 1);
+            jumpInButton.gameObject.SetActive(incompletedTask != null && !string.IsNullOrEmpty(incompletedTask?.coordinates));
+            jumpInDelegate = () => WebInterface.SendChatMessage(new ChatMessage
+            {
+                messageType = ChatMessage.Type.NONE,
+                recipient = string.Empty,
+                body = $"/goto {incompletedTask?.coordinates}",
+            });
+
+            readMoreDelegate = () => OnReadMoreClicked?.Invoke(quest.id);
+            questName.text = quest.name;
+            description.text = quest.description;
+            SetThumbnail(quest.thumbnail_entry);
+            pinQuestToggle.SetIsOnWithoutNotify(pinnedQuests.Contains(quest.id));
+
+            var questCompleted = quest.isCompleted;
+            pinQuestToggle.gameObject.SetActive(!questCompleted);
+            progressInTitle.fillAmount = quest.progress;
+            completedProgressInTitle.gameObject.SetActive(questCompleted);
+            completedMarkInTitle.gameObject.SetActive(questCompleted);
+        }
+
+        private void OnPinToggleValueChanged(bool isOn)
+        {
+            if (quest == null)
+                return;
+
+            if (!quest.canBePinned)
+            {
+                pinnedQuests.Remove(quest.id);
+                pinQuestToggle.SetIsOnWithoutNotify(false);
+                return;
+            }
+
+            if (isOn)
+            {
+                if (!pinnedQuests.Contains(quest.id))
+                    pinnedQuests.Add(quest.id);
+            }
+            else
+            {
+                pinnedQuests.Remove(quest.id);
+            }
+        }
+
+        private void OnPinnedQuests(string questId)
+        {
+            if (quest != null && quest.id == questId)
+                pinQuestToggle.SetIsOnWithoutNotify(true);
+        }
+
+        private void OnUnpinnedQuest(string questId)
+        {
+            if (quest != null && quest.id == questId)
+                pinQuestToggle.SetIsOnWithoutNotify(false);
+        }
+
+        internal void SetThumbnail(string thumbnailURL)
+        {
+            if (thumbnailPromise != null)
+            {
+                thumbnailPromise.ClearEvents();
+                AssetPromiseKeeper_Texture.i.Forget(thumbnailPromise);
+            }
+
+            if (string.IsNullOrEmpty(thumbnailURL))
+                return;
+
+            thumbnailPromise = new AssetPromise_Texture(thumbnailURL);
+            thumbnailPromise.OnSuccessEvent += OnThumbnailReady;
+            thumbnailPromise.OnFailEvent += x => { Debug.Log($"Error downloading quest panel entry thumbnail: {thumbnailURL}"); };
```

Comment:
Wouldn't it be more correct to use a `Debug.LogError()` here?

```suggestion
            thumbnailPromise.OnFailEvent += x => { Debug.LogError($"Error downloading quest panel entry thumbnail: {thumbnailURL}"); };
```